.github/workflows/github-pages-docs.yaml

```
@@ -0,0 +1,36 @@
+name: GitHub Pages
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - 'v*.*.*'
+  pull_request:
+
+jobs:
+  deploy:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: write
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+
+      - run: yarn install --frozen-lockfile # optional, --immutable
+      - run: yarn generate:docs
+
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v4
+        # TODO: uncomment this before merge
+        # if: github.ref == 'refs/heads/main'
```

TODO - this will guard against whatever PRs deploying.
I want to test deploy so this is temp disabled

```suggestion
        if: github.ref == 'refs/heads/main'
```